### PR TITLE
DTS SPI device, DW and quark_se

### PR DESCRIPTION
**Pull request by tbursztyka**

### CODE REVIEW

File: boards/x86/quark_se_c1000_devboard/quark_se_c1000_devboard.dts [hunk 1]
```diff
@@ -54,4 +54,12 @@
...
 &spi1 {
 	status = "ok";
+
+	cc2520@0 {
+		 compatible = "ti,cc2520";
+		 reg = <0x0>;
+		 spi-max-frequency = <8000000>;
+		 cs-gpios = <&gpio0 11 0>;
+		 label = "CC2520";
+	};
```
Comment (tbursztyka) [inline boards/x86/quark_se_c1000_devboard/quark_se_c1000_devboard.dts]: @lpereira I guess that's what you would be interested to, as we talked in device driver API meeting

File: dts/x86/intel_curie.dtsi [hunk 2]
```diff
@@ -136,5 +136,33 @@
...
+		spi0: spi@0xb0001000 {
+			compatible = "snps,designware-spi";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0xb0001000 0x400>;
+			label = "SPI_0";
+			interrupts = <2 IRQ_TYPE_LEVEL_HIGH 2>;
+			interrupt-parent = <&intc>;
+			interrupt-masq = <0xb0800454>;
+			fifo-depth = <7>;
```
Comment (galak) [inline dts/x86/intel_curie.dtsi]: where is fifo-depth specified?  dts binding?

File: dts/x86/intel_curie.dtsi [hunk 3]
```diff
@@ -136,5 +136,33 @@
...
+
+		spi1: spi@0xb0001400 {
+			compatible = "snps,designware-spi";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0xb0001400 0x400>;
+			label = "SPI_1";
+			interrupts = <3 IRQ_TYPE_LEVEL_HIGH 2>;
+			interrupt-parent = <&intc>;
+			interrupt-masq = <0xb0800458>;
```
Comment (galak) [inline dts/x86/intel_curie.dtsi]: where is interrupt-masq defined or used (what binding specifies this?)

**Review by galak — CHANGES_REQUESTED** [review]
So some of my comments about fifo-depth & interrupt-masq might be cause of github putting the patches in a weird order.

However, the cs-gpio change out of the controller node for SPI isn't acceptable.  The spi controller binding is specified here:

https://git.kernel.org/pub/scm/linux/kernel/git/devicetree/devicetree-rebasing.git/tree/Bindings/spi/spi-bus.txt

And we should conform to it, in which the CS are specified in the controller node, not in the device nodes.

File: dts/bindings/spi/spi-device.yaml [hunk 4]
```diff
@@ -29,6 +29,11 @@ properties:
       category: required
       description: Maximum clock frequency of device's SPI interface in Hz
       generation: define
+    cs-gpios:
```
[the remark this answers is not in the document]
Comment (tbursztyka) [reply dts/bindings/spi/spi-device.yaml]: No, see my commit message about removing cs-gpios from spi.yaml.

File: dts/x86/intel_curie.dtsi (hunk 3 above)
Comment (tbursztyka) [reply dts/x86/intel_curie.dtsi]: there is a patch adding it.

**Comment by tbursztyka** [comment]
Ok, this cs-gpios from linux is quite dumb to follow on Zephyr, as it's not the controller which defines it but the spi device. This makes it really counter intuitive.

File: dts/x86/intel_curie.dtsi (hunk 2 above)
Comment (tbursztyka) [reply dts/x86/intel_curie.dtsi]: hum, left over from previous rebase. have to remove that

**Comment by tbursztyka** [comment]
github really orders patch weirdly indeed.

**Comment by galak** [comment]
> Ok, this cs-gpios from linux is quite dumb to follow on Zephyr, as it's not the controller which defines it but the spi device. This makes it really counter intuitive.

While its a little counter intuitive, it makes sense from a device / device tree point of view.  Since a SPI device doesn't know anything about the controller its connected to it has no idea how the chip selects are implemented -- this is a function of the controller, and thus should live in the controller node.  Otherwise we'd have to update each binding for every combination of chip selection solutions that might exist for all controllers.

**Comment by tbursztyka** [comment]
@galak I reverted back to linux style, and remove the fifo stuff. Can you re-review?

File: dts/x86/intel_curie.dtsi [hunk 5]
```diff
@@ -136,5 +136,31 @@
 
 			status = "disabled";
 		};
+
+		spi0: spi@0xb0001000 {
```
Comment (galak) [inline dts/x86/intel_curie.dtsi]: spi@0xb0001000 -> spi@b0001000

File: dts/x86/intel_curie.dtsi [hunk 6]
```diff
@@ -136,5 +136,31 @@
...
+			reg = <0xb0001000 0x400>;
+			label = "SPI_0";
+			interrupts = <2 IRQ_TYPE_LEVEL_HIGH 2>;
+			interrupt-parent = <&intc>;
+			interrupt-masq = <0xb0800454>;
+
+			status = "disabled";
+		};
+
+		spi1: spi@0xb0001400 {
```
Comment (galak) [inline dts/x86/intel_curie.dtsi]: similar comment:

spi@0xb0001400 -> spi@b0001400

File: dts/bindings/spi/snps,designware-spi.yaml [hunk 7]
```diff
@@ -32,4 +32,10 @@ properties:
       category: required
       description: required interrupts
       generation: define
+
+    interrupt-masq:
```
Comment (galak) [inline dts/bindings/spi/snps,designware-spi.yaml]: I dont see this used anywhere in this patch set.  Can we drop it for now, I'm still confused what this is suppose to be used for.

File: samples/net/echo_client/sample.yaml [hunk 8]
```diff
@@ -16,8 +16,8 @@ tests:
     extra_args: CONF_FILE="prj_bt.conf"
     platform_whitelist: qemu_x86
     tags: bluetooth
-  test_cc2520:
-    extra_args: CONF_FILE="prj_frdm_k64f_cc2520.conf"
+  test_mcr20a:
+    extra_args: CONF_FILE="prj_frdm_k64f_mcr20a.conf"
```
Comment (galak) [inline samples/net/echo_client/sample.yaml]: This seems like we should have any entry for prj_frdm_k64f_cc2520.conf and one for prj_frdm_k64f_mcr20a.conf

**Review by galak — CHANGES_REQUESTED** [review]
Looks pretty good a few comments:

1. lets drop interrupt_masq for now since its not used
2. Fix '0x' in dts node names
3. Seems like the change to sample.yaml should support both mcr20a and cc2520 on k64f.

File: samples/net/echo_client/sample.yaml (hunk 8 above)
Comment (galak) [reply samples/net/echo_client/sample.yaml]: If we really are removing support for prj_frdm_k64f_cc2520.conf, we should remove the file as well.

File: dts/bindings/spi/snps,designware-spi.yaml (hunk 7 above)
Comment (tbursztyka) [reply dts/bindings/spi/snps,designware-spi.yaml]: How come you don't see it? Is github messing up things or?

See "dts/x86: Add the 2 master SPI port on curie SoC"
and "boards/x86: Enable usage of SPI definition through dts"

File: samples/net/echo_client/sample.yaml (hunk 8 above)
Comment (tbursztyka) [reply samples/net/echo_client/sample.yaml]: We could indeed. Btw: how to apply dts in samples?

File: dts/bindings/spi/snps,designware-spi.yaml (hunk 7 above)
Comment (galak) [reply dts/bindings/spi/snps,designware-spi.yaml]: I see it now, I wasn't tracing the #define's enough.

File: dts/x86/intel_curie.dtsi (hunk 3 above)
Comment (galak) [reply dts/x86/intel_curie.dtsi]: So interrupt-masq should be a second element in the reg property.  As this is describing a register region that needs access.  You can use 'reg-names' if you want  to have names for the properties as well.  So something like:

```
reg = <0xb0001400 0x400
            0xb0800458 0x4>;
reg-names = <"base" "irq-masq">;
```

**Review by galak — CHANGES_REQUESTED** [review]
Now that I understand a bit more about what interrupt-masq is trying to accomplish, we should do that via the reg property, see inline comment.

Also, from the driver point of view this seems to be required, or at least CONFIG_SPI_DW_PORT_n_INT_MASK is required.  So not sure if the code should be:

#ifdef CONFIG_SPI_DW_PORT_n_INT_MASK
_spi_int_mask(CONFIG_SPI_DW_PORT_n_INT_MASK)
#endif

**Comment by pfalcon** [comment]
Added @MaureenHelm and @dleach02 to be in loop on frdm_k64f changes. 

(Removed myself from reviewers as I can't say much on the changes.)

**Comment by galak** [comment]
@tbursztyka any plans to address comments?

**Comment by galak** [comment]
> @tbursztyka any plans to address comments?

Any plans?

**Review by galak — CHANGES_REQUESTED** [review]
n/a, updated wrong PR

**Comment by nashif** [comment]
this was covered in #10249

**Pull request closed without merging**